Assets/Scripts/Gameplay/ConnectionManagement/ConnectionState/ClientConnectedState.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
+
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    /// <summary>
+    /// Connection state corresponding to a connected client. When being disconnected or timed out, transitions to the
+    /// Offline or Reconnecting state, depending on the cause of the disconnect.
+    /// </summary>
+    public class ClientConnectedState : ConnectionState
+    {
+        IPublisher<ConnectStatus> m_ConnectStatusPublisher;
+
+        [Inject]
+        void InjectDependencies(IPublisher<ConnectStatus> connectStatusPublisher)
+        {
+            m_ConnectStatusPublisher = connectStatusPublisher;
+        }
+
+        public override void Enter() { }
+
+        public override void Exit() { }
+
+        public override void OnClientDisconnect(ulong clientId)
+        {
+            m_ConnectStatusPublisher.Publish(ConnectStatus.Reconnecting);
+            m_ConnectionManager.ChangeState(ClientReconnecting);
+        }
+
+        public override void OnUserRequestedShutdown()
```

Comment:
As for the publishing of UserRequestedDisconnect status, it could be moved to the base class, or handled directly by the ConnectionManager itself. Would that make sense?

Reply:
my thinking is mostly about making sure everyone has the same "procedure" for shutting things down. This could be in a static method in a utils as well.